### PR TITLE
fix: test folder structure for asset-compute generator

### LINKED ISSUE #157: asset-compute worker test file structure is missing required folders
### Expected Behaviour
When creating a project with Asset Compute Workers, the expected test file structure is this:
```
project/
  test/asset-compute/
        worker-1/
             <test-case>/
        worker-2/
             <test-case>/
```
(see example here: https://github.com/adobe/asset-compute-example-workers/tree/master/projects/worker-azure-ai/test/asset-compute)

The test files need to be attached to workers since they only apply to specific workers, not all the workers in the project. Also, our [aio-cli-plugin-asset-compute](https://github.com/adobe/aio-cli-plugin-asset-compute/blob/master/src/commands/asset-compute/test-worker.js#L23) expects this specific file structure to be able to find the tests.
### Actual Behaviour
The new file structure is not the same, the test cases are not nested in the correct folders: `asset-compute/<worker-name>` which means the tests are not connected to specific workers. Also, the `test` and `actions` files are nested under `src/dx-asset-compute-worker-1` instead of being in the root of the project.

new-aio-structure

So when you run `aio app test`, our plugin cannot find the test files:
```
[jamiedelbick:...gration-tests/build/project]$ aio app test                                                             
Using local project's @adobe/aio-cli-plugin-asset-compute version 1.5.0dx/asset-compute/worker/1/* (hook) - running tests...
No worker tests found in 'test/asset-compute/*'app:test results:
    √ dx/asset-compute/worker/1/* (hook)
```

#### Steps to Reproduce
1. install latests version of aio-cli (v8) and dependents
2. `aio app init`
3. (follow steps to choose asset compute worker)
4. run `aio app test`

#### Platform and Version
```
Using local project's @adobe/aio-cli-plugin-asset-compute version 1.5.0


  System:
    OS: macOS 11.2.3
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 20.04 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 12.20.1 - ~/.nvm/versions/node/v12.20.1/bin/node
    Yarn: 1.22.5 - /usr/local/bin/yarn
    npm: 6.14.10 - ~/.nvm/versions/node/v12.20.1/bin/npm
  Virtualization:
    Docker: 20.10.7 - /usr/local/bin/docker
  npmGlobalPackages:
    @adobe/aio-cli: 8.0.0

  CLI plugins:
    core:
      @adobe/aio-cli 8.0.0
      @adobe/aio-cli-plugin-app 8.0.0
      @adobe/aio-cli-plugin-auth 2.4.1
      @adobe/aio-cli-plugin-config 2.3.0
      @adobe/aio-cli-plugin-console 3.4.0
      @oclif/plugin-autocomplete 0.3.0
      @oclif/plugin-help 2.2.3
      @oclif/plugin-not-found 1.2.4
      @oclif/plugin-plugins 1.10.1
      @oclif/plugin-warn-if-update-available 1.7.0
    user:
      @adobe/aio-cli-plugin-asset-compute 1.5.0
      @adobe/aio-cli-plugin-certificate 0.3.0 (*)
      @adobe/aio-cli-plugin-events 1.1.5 (*)
      @adobe/aio-cli-plugin-info 2.0.0 (*)
      @adobe/aio-cli-plugin-runtime 4.0.0 (*)
    link:
```

#### Sample Code that illustrates the problem
There are two issues:
1. asset compute worker tests cases should be nested in two crutial folders: `test/asset-compute/<worker-name>`
I believe this change was brought in through this PR: https://github.com/adobe/generator-aio-app/pull/148/files#diff-edddb1be7fc41fd5d024cfea1b932ed6cc4fc9344bf0bc66de503c0f0a7a8d55L80-L88
removed-asset-compute-folder

2. new aio file structure that makes the test folder nested inside `src/dx-asset-compute-worker-1`.

This is a critical bug because `aio app test` is broken for all Asset Compute Workers created with the most recent version of the aio cli

### PR DESCRIPTION
**Pull request by shazron**
Fixes #157 

## Motivation and Context

Minimal changes to the expected structure for asset-compute tests.

The asset-compute tests expect a certain location for the tests relative to the root:
```
|____test
| |____asset-compute
| | |____worker
| | | |____simple-test
| | | | |____file.jpg
| | | | |____rendition.jpg
| | | | |____params.json
| | | |____corrupt-input
| | | | |____file.jpg
| | | | |____params.json
|____actions
| |____worker
| | |____index.js
|____ext.config.yaml
```
(here `worker` is the name of the action)

⚠️ However, now the root is not the root of the app, but the root of the extension.

## How Has This Been Tested?

1. aio-cli@8 
2. `aio app init` (get the asset compute extension)
3. Modified https://github.com/adobe/aio-cli-plugin-asset-compute/blob/a13a83ab5a6ea28b35e4fa80a05c3a0beaca9735/src/commands/asset-compute/test-worker.js#L23 with `const TEST_DIR = path.join("src", "dx-asset-compute-worker-1", "test", "asset-compute");`
4. aio app test

The tests run, but there are errors in the `asset-compute` plugin  (which are outside the scope of the generator) due to the new extension registry changes.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by moritzraho — APPROVED** [review]

**Comment by moritzraho** [comment]
@shazron can we set the tests config in <ext|app>.config.yaml  to point to the `asset-compute` test folder ? that would help for delete commands

File: test/generators/add-action/asset-compute.test.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ describe('prototype', () => {
 
 function assertGeneratedFiles (actionName) {
   const actionPath = `actions/${actionName}`
-  const testPath = 'test'
+  const testPath = `test/asset-compute/${actionName}`
```
Comment (jdelbick) [inline test/generators/add-action/asset-compute.test.js:44]: is it possible to have these tests actually run `aio app test`? (in the future after the aio cli plugin asset-compute works with the new aio)

I think that would have helped catch this bug earlier

Comment (shazron) [reply test/generators/add-action/asset-compute.test.js:44]: Not sure I follow exactly. Yes, if the tests are jest tests, these can be run automatically by `aio app test`. Right now it reads the `test` hook in the extension's `ext.config.yaml` and runs that (this is analogous to the old way where it ran your test via adding the command to `npm test`)

Comment (jdelbick) [reply test/generators/add-action/asset-compute.test.js:44]: Yes that was what I was picturing, running `aio app test` on the created project

**Pull request merged**